src/shared_memory_manager.cc

```diff
@@ -264,6 +264,45 @@ OpenCudaIPCRegion(
 
   return nullptr;
 }
+
+TRITONSERVER_Error*
+GetCudaSharedMemoryRegionSize(CUdeviceptr data_ptr, size_t& shm_region_size)
+{
+  CUdeviceptr* base = nullptr;
+  CUresult result = cuMemGetAddressRange(base, &shm_region_size, data_ptr);
+  if (result != CUDA_SUCCESS) {
+    const char* errorString;
+    if (cuGetErrorString(result, &errorString) != CUDA_SUCCESS) {
+      return TRITONSERVER_ErrorNew(
+          TRITONSERVER_ERROR_INTERNAL, "Failed to get CUDA error string");
+    }
+    // Should not pass the detailed error message back to the client.
+    LOG_ERROR << "Failed to get CUDA address range: " << errorString;
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_INTERNAL, "Failed to get CUDA address range");
+  }
+  return nullptr;
+}
+
+TRITONSERVER_Error*
+CheckCudaSharedMemoryRegionSize(
+    const std::string& name, CUdeviceptr data_ptr, size_t byte_size)
+{
+  size_t shm_region_size = 0;
+  RETURN_IF_ERR(GetCudaSharedMemoryRegionSize(data_ptr, shm_region_size));
```

Comment:
instead of return if error - we could treat this as internal - 

I think I would just do
```
success = GetSharedMemoryRegionSice()
if not sucesss or (byte_size > shm_region_size)
{
 return "Failed to register shared memory region .... "
}
```

Reply:
that way we have one consistent simple error message (details of why it failed are not so relevant to client in this case) - and we have logs for more detailed analysis.

Reply:
Makes sense! Do you think we'd still need to keep the generic error message if we are only returning a single error message for "failed to register shared memory .. invalid args"?
```
return TRITONSERVER_ErrorNew(
        TRITONSERVER_ERROR_INTERNAL, "Failed to get shared memory information");
```
I think we could probably do something like
```
{
   // In GetCudaSharedMemoryRegionSize
   // Return the details with Triton error object, which will NOT be exposed to client
   return TRITONSERVER_ErrorNew(
             TRITONSERVER_ERROR_INTERNAL, "Failed to get CUDA error string");
}
...
...
{
   auto err = GetCudaSharedMemoryRegionSize(data_ptr, shm_region_size);
   // User-provided offset and byte_size should not go out-of-bounds.
   if (err != nullptr || byte_size > shm_region_size) {
    // Should not pass the detailed error message back to the client.
    LOG_ERROR << TRITONSERVER_ErrorMessage(err); // log the error in one place
    return TRITONSERVER_ErrorNew(
        TRITONSERVER_ERROR_INVALID_ARG,
        std::string(
            "failed to register shared memory region '" + name +
            "': invalid args")
            .c_str());
}

```